### ISSUE TITLE
Wrist Pain!

### ISSUE THREAD

**Issue by mikeal** [mikeal/self-care#1]
I've had intermittent wrist pain for years, often I sort of "injur" a wrist and for a month or so even small things like pushing open doors will hurt.

I ended up getting some exercise bands and spending just a few minutes every other day doing basic wrist exercises and I haven't had any problems since.

**Comment by ozantunca** [comment]
I have a similar issue with my wrist. Things started going down for me after I started feeling extreme pains every time I tried to do push ups. Even after I gave up push ups, the problem continued and I started having “injuries” exactly like you described. Eventually I went to a doctor and got an MRI scan which revealed a sort of cyst in my wrist called a gonglion cyst. It’s not dangerous but having a surgery to remove it could be dangerous so I left it there. Instead, I started doing some weight lifting every now and then to strengthen my wrist and it works pretty great.

**Comment by davidmarkclements** [comment]
After speaking to a physio recently about pain across the back of the hand, he made a point that it can be referred pain - e.g the problem might actually be in your back/shoulder - or along any of the nerve paths leading to the hand. Exercising twice a day and stretching my back, arms, sides and legs (tight legs also pull on the back), twice a day definitely makes a difference for me (although it’s more of a bonus, I have to maintain this routine for other reasons anyway)

**Comment by jreina** [comment]
I recommend an ergo keyboard to all of my friends. Once the wrist pain starts, it is very hard to get rid of. I use a Microsoft Sculpt at work and I notice that my mech keyboard at home makes typing unbearable. It took some time to get used to the Sculpt but it was very much worth the effort.

I recently had a friend complain about their wrist and shoulder pain. Once they switched to an ergo keyboard the pain stopped and has not returned.

I know everyone is different with respect to pain causes but please do not overlook ergo keyboards. I love fancy mechs (actually a bit of a mech junkie) but your health absolutely needs to come first.

**Comment by ilias-t** [comment]
Yeah I had really bad elbow pain and transitioning to a split [keyboard](https://www.amazon.com/Kinesis-Freestyle2-Ergonomic-Keyboard-Separation/dp/B009ZNBJK8) made a massive difference.

**Comment by TehShrike** [comment]
I've tried to be pretty sensitive to wrist pain.  I could feel it reaching a dangerous point ~8 years ago.

I started using [trackball](https://www.amazon.com/gp/product/B016QCPO0G/) at work instead of a mouse, and that one step was enough to reduce the pain I'd been feeling.

I have taken other steps since then, including using a wrist rest for my keyboard anv eventually buying a [split keyboard with split wrist rests](https://ultimatehackingkeyboard.com/) (still a fancy mechanical keyboard!), which does reduce stress, but getting away from mice was my biggest breakthrough.

**Comment by rwaldron** [comment]
I was experiencing similar wrist pain several years ago. In March of 2015, I started going to the gym twice a week to get some exercise: a two mile jog, sit-ups and pushups, and then weight lifting. After a few weeks the pain was gone and I haven't felt it since then.

**Comment by coolaj86** [comment]
# Dvorak

I don't use it for speed. I literally use it to alleviate wrist pain. It's very intuitive and only took me an hour to memorize the individual letters (literally could type without looking at the keyboard within an hour), however, it took about 3 weeks to re-learn the muscle memory of whole words and common patterns.

# BetterTouchTool & Karabiner Elements

Remap your keys

* Delete -> Caps Lock
* Right CMD -> Enter
* Right ALT -> Delete
* Shift+CMD + I/J/K/L -> Up/Left/Down/Right
* (or swap to the left side and W/A/S/D if you typically use your right hand for controls)

Basically any time you experience wrist pain you identify the movement and then remap the keys to a location that requires less wrist movement.

# Magic Trackpad v1

If you don't have a Mac, get one. The ergonomics of the Trackpad are literally worth the extra $2k.

Mice just won't do for wrist pain. If you must use a mouse, learn to be ambidextrous (it's not that hard, but might take a week of regular use... unless you do graphics work, in which case it may not be reasonable to pursue).

Unfortunately no other operating system has proper trackpad drivers. You could maybe get it to work with a hackintosh, but without the native OS X drivers the Magic Trackpad has no wow-factor other than that it works at-all (which would be a big wow for the people that carry a mouse in their backpack to compensate for... every other trackpad).

Also, the v2 Trackpad kinda sucks. The response is more uniform across the entire surface (top right same as bottom left or middle/center), but they use haptic feedback with a starkly noticeable delay rather than real clicking.

# MacBook 2012-2015

From eyes to fingers (and maybe even toes), the MacBook is just more ergonomic.

But maybe stick with the 2012-2015 models (before the haptic feedback replaced the manual click on the trackpad). If you don't mind a full 100ms of delay between the time you click and the time the buzzer simulates the click maybe that won't bother you, but I can't stand physical presses being out of sync with the felt experience.

**Comment by murtraja** [comment]
I had this recently, what I tried to do was keep a soft cushion like a handkerchief just below my wrist while using the keyboard, this helped me heal and mitigate the pain.

**Comment by mikeal** [comment]
I used to get strain in my write from trackpad usage which went away when I migrated to the iPad Pro for development (which is a much longer and more involved post). After migrating I have to say that the mouse feels like a legacy interface that, ergonomically, was always problematic.

**Comment by coolaj86** [comment]
(not trying to clutter the comments here, but I'd like to read that post)

**Comment by TehShrike** [comment]
@mikeal what bands do you use/exercises do you do?

**Comment by mikeal** [comment]
I got these ones from amazon and use the “Light” band. https://www.amazon.com/Fit-Simplify-Resistance-Exercise-Instruction/dp/B01AVDVHTI

**Comment by TehShrike** [comment]
What wrist exercises did you find useful?  I did some brief googling and found lots of random finger/hand stuff and couldn't tell which ones were meaningful.

Also using this new MBP in coffee shops is giving me RSI so I need to start something soon :-|

**Comment by MikeMcQuaid** [comment]
In addition to the suggestions above (going to the gym helped me a lot) my wife is a physical therapist/physiotherapist so has helped me a bunch with ergonomics. I've found my wrist position/comfort is heavily affected by chair/desk height and keyboard width so use a adjustable sit/stand desk, adjustable chair and a narrow keyboard so my mouse/trackpad can be nearer.
